### ISSUE TITLE
workbox-webpack-plugin - InjectManifest links to manifest relative to webpack root, not the swDest

### ISSUE THREAD

**Issue by devnook** [GoogleChrome/workbox#1543]
**Library Affected**:
workbox-webpack-plugin

If I use swDest option in webpack plugin, InjectManifest creates a link in injectScripts in SW that is relative to webpack root, not the new swDest. Therefore, the injectScripts call in SW cannot find the manifest file. 

Webpack config:

```
const { InjectManifest } = require('workbox-webpack-plugin');

module.exports = {
  entry: {
    main: './src/client/js/app.js',
    ...
  },
  output: {
    filename: 'js/[name].js',
    chunkFilename: 'js/chunk-[name].js',
    path: path.resolve(__dirname, 'dist/static/js'),
  },
  plugins: [
    new InjectManifest({
      swSrc: './src/sw.js',
      swDest: '../sw.js',
    })
  ]
};

```
Outcome dir structure:

```
|--static
...... |-sw.js 
...... |--js
............ |--main.js
............ |--precache-manifest.XXX.js
```

Content of sw.js:

`importScripts("precache-manifest.XXX.js", ...);   //<- the js subdir is missing
`
I can solve it by changing the whole webpack to output to /static, but then swDest is obsolete.

**Comment by chippison** [comment]
I also have this problem with GenerateSW.
This happens when swDest is outside of webpack root directory

**Comment by jeffposnick** [comment]
I dug into this a bit more, and it's not going to be trivial to support `swDest` locations outside of the to `output.path` (at least not with the current codebase).

What we can do is start warning developers when they configure things that way rather than just silently writing a `swDest` with incorrect paths.

Taking a step back, though, @devnook, this strikes me as an odd webpack config:

```js
output: {
  filename: 'js/[name].js',
  chunkFilename: 'js/chunk-[name].js',
  path: path.resolve(__dirname, 'dist/static/js'),
},
```

Your `output.path` value ends in `js`, and your `output.filename` value starts with `js/`, so wouldn't that lead to a `js/js/` in your output directory structure?

E.g. when I run webpack using your sample config, I get output that looks like:

```sh
dist
└── static
    ├── js
    │   ├── js
    │   │   └── bundle.js
    │   └── precache-manifest.25cf57272b51992d973cc0b1d71f4c02.js
    └── sw.js
```

You write:

> I can solve it by changing the whole webpack to output to /static, but then swDest is obsolete.

But I think that sounds like what you should do, i.e. make `output.path` set equal to the root of your web app, and then use subdirectory as needed in your `filename` values.

If I go with the following webpack config, I think you can get what you're looking for without any changes to Workbox:

```js
output: {
  filename: path.join('js', '[name].js'),
  chunkFilename: path.join('js', 'chunk-[name].js'),
  path: path.resolve(__dirname, 'dist', 'static'),
},
```

That leads to the following output, and all of the internal relative paths are set correctly inside of the Workbox-generated code:

```sh
dist
└── static
    ├── js
    │   └── bundle.js
    ├── precache-manifest.25cf57272b51992d973cc0b1d71f4c02.js
    └── sw.js
```

Can you try making that change?

**Comment by jeffposnick** [comment]
This should be addressed by the current Workbox v5.0.0 alpha.
